modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/DropwizardServerGenerator.scala

```diff
@@ -287,21 +287,33 @@ object DropwizardServerGenerator {
                 method.addAnnotation(new SingleMemberAnnotationExpr(new Name("Path"), new StringLiteralExpr(pathSuffix)))
               }
 
-              val allConsumes = operation.get.consumes.flatMap(ContentType.unapply).toList
+              val allConsumes = operation.downField("consumes", _.consumes).map(_.flatMap(ContentType.unapply)).unwrapTracker
               val consumes    = DropwizardHelpers.getBestConsumes(operation, allConsumes, parameters)
               consumes
                 .map(c => new SingleMemberAnnotationExpr(new Name("Consumes"), c.toJaxRsAnnotationName))
                 .foreach(method.addAnnotation)
-              val allProduces = operation.get.produces.flatMap(ContentType.unapply).toList
-              val bestSuccessResponse = responses.value
-                .filter(_.statusCode / 100 == 2)
-                .find(_.value.isDefined)
-              val produces = bestSuccessResponse.flatMap(
-                DropwizardHelpers.getBestProduces(operationId, allProduces, _)
-              )
-              produces
-                .map(p => new SingleMemberAnnotationExpr(new Name("Produces"), p.toJaxRsAnnotationName))
-                .foreach(method.addAnnotation)
+
+              val allProduces = operation.downField("produces", _.produces).map(_.flatMap(ContentType.unapply)).unwrapTracker
+              NonEmptyList
+                .fromList(
+                  responses.value
+                    .flatMap(DropwizardHelpers.getBestProduces(operationId, allProduces, _))
+                    .distinct
+                    .map(_.toJaxRsAnnotationName)
+                )
+                .foreach(
```

Comment:
What does it mean to have more than one `@Produces`? Is that legit?

Reply:
Yeah, so I sorta mentioned this in DM on slack, but it is indeed allowed -- well, rather, you have one @Produces, but it can hold a `String[]` rather than just `String`.

Unfortunately, this doesn't currently work because the old swagger 2-to-3 `.produces` shim causes us to lose which content type is associated with which response.  That'll require some extra plumbing to fix, but I'll handle that in another PR.